filters/accesslog/control.go

```diff
@@ -18,13 +21,22 @@ const (
 
 // Common filter struct for holding access log state
 type AccessLogFilter struct {
-	Enable   bool
+	// Enable represents whether or not the access log is enabled.
+	Enable bool
+	// Prefixes contains the list of response code prefixes.
 	Prefixes []int
+	// MaskedQueryParams contains the set of query parameters (keys) that are masked/obfuscated in the access log.
+	MaskedQueryParams map[string]struct{}
 }
 
 func (al *AccessLogFilter) Request(ctx filters.FilterContext) {
 	bag := ctx.StateBag()
-	bag[AccessLogEnabledKey] = al
+	if f, ok := bag[AccessLogEnabledKey]; ok {
+		a := f.(*AccessLogFilter)
+		maps.Copy(a.MaskedQueryParams, al.MaskedQueryParams)
```

Comment:
This modifies filter instance from parallel requests and will panic.

Reply:
@AlexanderYastrebov seems that I miss some knowlege here.
How could this modification cross the boundary between requests if the state bag is unique to each request and filters are executed sequentially?
https://github.com/zalando/skipper/blob/46828d43055dab3a93a37e21fddab44b49bfe5ce/filters/filters.go#L85-L87

Reply:
nvm, I got it 🤦